test/test_integration_ssl.rb

```diff
@@ -189,6 +188,15 @@ def test_ssl_run_with_localhost_authority
     end
   end
 
+  def test_ssl_run_with_localhost_authority_in_cluster_mode; require "open3"
+    skip_if :jruby
+
+    cmd = "#{BASE} bin/puma -b \"ssl://0.0.0.0:#{bind_port}\" -t 5:5 -w 2 test/rackup/self_signed.ru"
+    _stdin, _stdout, stderr, _wait_thr = Open3.popen3(cmd)
```

Comment:
I think @MSP-Greg just removed some other tests using `Open3` (because they were unreliable in CI) so we should try to not introduce more. I'm sure it is possible to test this in some other way, but I can't say exactly how right now. Maybe someone else can chime in on that or if you explore existing test you can find inspiration. 

Also not sure it needs a test? Aren't there any for the empty `key`/`key_pem` scenario?

Reply:
I rewrote the test. Please check it